### PR TITLE
DDCE-4866 Fix disabled Role options

### PR DESCRIPTION
**Pull request by mywyau**

### CODE REVIEW

File: app/viewmodels/application/RoleDetailsSummary.scala [hunk 1]
```diff
@@ -17,7 +17,9 @@
 package viewmodels.application
 
 import models.requests.WhatIsYourRole
+import models.{CheckMode, DraftId}
```
Comment (gastonracui) [inline app/viewmodels/application/RoleDetailsSummary.scala:20]: optimize imports

File: app/views/components/yesNoRadio.scala.html [hunk 2]
```diff
@@ -0,0 +1,58 @@
+@*
+ * Copyright 2022 HM Revenue & Customs
```
Comment (gastonracui) [inline app/views/components/yesNoRadio.scala.html:2]: 2024...

File: test/models/ModeSpec.scala [hunk 3]
```diff
@@ -29,8 +29,8 @@ class ModeSpec extends SpecBase {
       Mode.jsLiteral.to(CheckMode) mustBe "CheckMode"
     }
 
-    "return a read-only mode type" in {
-      Mode.jsLiteral.to(ReadOnlyMode) mustBe "ReadOnlyMode"
+    "return a unchanged mode type" in {
```
Comment (gastonracui) [inline test/models/ModeSpec.scala:32]: return an unchanged

**Review by gastonracui — CHANGES_REQUESTED** [review]

**Review by gastonracui — APPROVED** [review]

**Review by gastonracui — APPROVED** [review]

File: app/views/components/Caption.scala.html [hunk 4]
```diff
@@ -16,6 +16,5 @@
 
 @this()
 
-@(message: String, captionPrefix: String="caption.prefix")(implicit messages: Messages)
-<h2 class="govuk-caption-xl"><span class="govuk-visually-hidden">
-    @messages(captionPrefix)</span>@message</h2>
```
Comment (EdCope) [inline app/views/components/Caption.scala.html:21]: @gastonracui @mywyau Was this change confirmed? I can't see any comments on the ticket about removing the `caption.prefix`

Comment (mywyau) [reply app/views/components/Caption.scala.html:21]: Hmm no don't think this should of changed. My mistake will revert and run tests

Comment (EdCope) [reply app/views/components/Caption.scala.html:21]: I don't think you need to revert. I can add it back in for our rebase change.

**Pull request merged**